src/main/java/de/unijena/cheminf/mortar/gui/controls/ToggleSwitch.java

```diff
@@ -0,0 +1,309 @@
+    private static final int RECTANGLE_WIDTH_VALUE = 45;
+    /**
+     * Default value for the width of the Background.
+     */
+    private static final int RECTANGLE_HEIGHT_VALUE = 18;
+    /**
+     * Default value for the radius of the Button.
+     */
+    private static final int CIRCLE_RADIUS_VALUE = 10;
+    /**
+     * Default value for the Layout which sets the position of the Background of the switch on the x-axis.
+     */
+    private static final int RECTANGLE_POSITION_VALUE = -50;
+    /**
+     * Default value for the Layout which sets the position of the button on the x-axis.
+     */
+    private static final int CIRCLE_POSITION_X_VALUE = -40;
+    /**
+     * Default value for the position of the Button on the y-axis.
+     */
+    private static final int CIRCLE_POSITION_Y_VALUE = 9;
+    /**
+     * Default value for the radius of the shadow on the button.
+     */
+    private static final int CIRCLE_SHADOW_RADIUS = 5;
+    /**
+     * Default value for the duration of the animated transition in seconds.
+     */
+    private static final double DURATION_VALUE = 0.25;
+    /**
+     * Default color of the background when the toggle switch is turned off.
+     */
+    private static final Color RECTANGLE_COLOR_OFF = Color.LIGHTGRAY;
+    /**
+     * Default color for the outline of the Background.
+     */
+    private static final Color RECTANGLE_OUTLINE_COLOR = Color.DARKGRAY;
+    /**
+     * Default color of the button.
+     */
+    private static final Color CIRCLE_COLOR = Color.WHITE;
+    /**
+     * Default color for the outline of the button.
+     */
+    private static final Color CIRCLE_OUTLINE_COLOR = Color.DARKGRAY;
+    /**
+     * Default color for the background when turned on.
+     */
+    private static final Color RECTANGLE_COLOR_ON = Color.web("#0099cc");
+    /**
+     * Default color of the shadow on the button.
+     */
+    private static final Color CIRCLE_SHADOW_COLOR = Color.GRAY;
+    /**
+     * Default boolean state of the switch.
+     */
+    private static final boolean SWITCH_STATE = false;
+    //</editor-fold>
+    //<editor-fold desc="Constructor with default values" defaultstate="collapsed">
+    /**
+     * Constructor.
+     * The toggle switch is built by initializing a layout which includes the sizing of the background (a rectangle)
+     * and the sizing of the button (a circle), as well as their colors. The transition of the button is initialized by
+     * adding an animation which also entails a color change. The transition works by subtracting the diameter of the
+     * circle from the width of the rectangle. A listener and a mouse listener trigger the animation
+     * when the button (circle) is clicked on. The rectangle height variable is used for its arc width and arc height,
+     * which provides the rounded shape of the switch.
+     */
+    public ToggleSwitch() {
+        super();
+        this.switchStateBooleanProperty = new SimpleBooleanProperty(SWITCH_STATE);
```

Comment:
Like the this-statement shows that a variable is an instance variable, public static variables can be explicitly declared with the class name in the lead, so "ToggleSwitch.SWITCH_STATE".